lib/resources/auditd_rules.rb

```diff
@@ -177,7 +177,7 @@ def get_action_list(line)
 
     # NB only in file lines
     def get_key(line)
-      line.match(/-k ([^ ]+)/)[1]
+      line.match(/-k ([^ ]+)/)[1] if line =~ /-k/
```

Comment:
I think we should be more specific about this condition. What is the filename in the line is `/etc/private-keys` but has no `-k SOMEKEY` indication? This will still match the `/-k/` condition.

Perhaps we should change that to: `if line.include?('-k ')` so we directly match on the trailing space? And add a line to your mock file that has a filename that includes `-k` in the file name?

Reply:
Sure thing...thanks!